src/librustc_mir/interpret/operand.rs

```diff
@@ -535,6 +537,7 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             }
             ty::ConstKind::Value(val_val) => val_val,
         };
+        let val_val = M::eval_maybe_thread_local_static_const(self, val_val)?;
```

Comment:
There is a `return` above, is it a problem that we do not run in that branch?

Reply:
That can never happen for TLS. by construction it is guaranteed that thread locals mentioned in MIR are already evaluated constants: https://github.com/rust-lang/rust/blob/534a41a32952d36ec73656357777ebbea707aeb4/src/librustc_mir_build/hair/cx/expr.rs#L757

Reply:
You know I don't like those answers relying on external invariants -- they should always be guarded by `assert!`. ;)

But `const_eval` actually calls `eval_const_to_op`, so we still go through this path. So all that is needd is adding a comment that explains that. :D (Also the function names are confusing...)

Reply:
I have added a comment that explains why `adjust_global_const` is guaranteed to be always called.